### ISSUE TITLE
[Monitoring V2 UI] PVs should not have a selector defined by default

### ISSUE THREAD

**Issue by aiyengar2** [rancher/dashboard#1757]
Related Issue: https://github.com/rancher/rancher/issues/29839

In the current Dashboard, we set `selector: { matchExpressions: [], matchLabels: {} }` when persistent storage is enabled, as shown here:
https://github.com/rancher/dashboard/blob/07e283e8198ad544e33b1cfda5e45f40bb7d1be0/chart/monitoring/prometheus/index.vue#L153

However, this field needs to be entirely omitted unless a user specifically chooses to set selectors since, as noted in 
https://github.com/rancher/rancher/issues/29839#issuecomment-719437251, `a PVC with a non-empty selector can't have a PV dynamically provisioned for it`. Since the current UI option by default make the selector non-empty, it breaks for provisioners like Longhorn by default.

#### Changes Requested

By default, the UI should show the following (e.g. no default rule shown for selectors)

![Screen Shot 2020-10-30 at 2 26 44 PM](https://user-images.githubusercontent.com/13127443/97758356-f2399a80-1abb-11eb-94fd-22c5b20ee3cd.png)

If no default rules are selected, the `prometheus.prometheusSpec.storageSpec.volumeClaimTemplate.selector` (and corresponding options for alertmanager) field should not exist on submit.

We should also add help text that says:
```
If you are using a dynamic provisioner (e.g. Longhorn), no Selectors should be specified since a PVC with a non-empty selector can't have a PV dynamically provisioned for it.
```

**Comment by jiaqiluo** [comment]
on rancher:v2.5-94397a0c5f79394e0e0f360e011593110934ddec-head with the latest UI

The bug itself is fixed, now we can enable the persistent storage for Prometheus by using the Longhorn as the storage class.  
![image](https://user-images.githubusercontent.com/6218999/97897312-893c6780-1cf3-11eb-8a24-64cfbe723c4d.png)
![image](https://user-images.githubusercontent.com/6218999/97897341-92c5cf80-1cf3-11eb-8211-d599ecf9bb8f.png)



But there is a small syntax issue in the warning message as  the following screenshot shows: 

<img width="1356" alt="Screen Shot 2020-11-02 at 9 56 00 AM" src="https://user-images.githubusercontent.com/6218999/97897187-5eeaaa00-1cf3-11eb-9d33-1f4d714850b6.png">

**Comment by westlywright** [comment]
@jiaqiluo Already in the PR :) https://github.com/rancher/dashboard/pull/1767

**Comment by jiaqiluo** [comment]
The fix is validated on rancher:`v2.5-100198aff107b9a0feccdcc3e6fb5b57e83f1fd7-head` with the latest UI

![image](https://user-images.githubusercontent.com/6218999/98017391-67f37e00-1dbc-11eb-91e2-f70fcf14adc3.png)

**Comment by ikogan** [comment]
We just experienced this with Rancher 2.6.10 and Rancher Monitoring 100.2.0+up40.1.2.
